### PR TITLE
apim vnet with nsg 

### PR DESCRIPTION
**Pull request by solankisamir**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Update Apim Version to 2018-01-01
* Add NSG according to documentation and ServiceTags

### CODE REVIEW

**Comment by solankisamir** [comment]
@bmoore-msft This template is failing as the value of `GEN-CUSTOM-DOMAIN-SSLCERT-PFXDATA`     starts with `-----BEGIN CERTIFICATE-----
MIIDLTCCAh -----END CERTIFICATE-----`

We only expect the value to have Base64 endcoded certificate there. How can we solve the issue for the CI?

File: 201-api-management-create-with-vnet/azuredeploy.json [hunk 1]
```diff
@@ -107,9 +71,413 @@
...
+              "provisioningState": "Succeeded",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "80",
+              "sourceAddressPrefix": "Internet",
+              "destinationAddressPrefix": "VirtualNetwork",
+              "access": "Allow",
+              "priority": 100,
+              "direction": "Inbound",
+              "sourcePortRanges": [],
```
Comment (bmoore-msft) [inline 201-api-management-create-with-vnet/azuredeploy.json]: remove emptyArrays on all of the rules

File: 201-api-management-create-with-vnet/azuredeploy.json [hunk 2]
```diff
@@ -107,9 +71,413 @@
...
+              "direction": "Outbound",
+              "sourcePortRanges": [],
+              "destinationPortRanges": [
+                "443",
+                "12000",
+                "1886"
+              ],
+              "sourceAddressPrefixes": [],
+              "destinationAddressPrefixes": [],
+              "destinationPortRange": null
```
Comment (bmoore-msft) [inline 201-api-management-create-with-vnet/azuredeploy.json]: remove all null properties

Comment (solankisamir) [reply 201-api-management-create-with-vnet/azuredeploy.json]: fixed

File: 201-api-management-create-with-vnet/azuredeploy.json (hunk 1 above)
Comment (solankisamir) [reply 201-api-management-create-with-vnet/azuredeploy.json]: fixed

**Comment by solankisamir** [comment]
@bmoore-msft can you take a look again?

**Pull request merged**